SonarQube.TeamBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -148,21 +141,17 @@ private bool FetchArgumentsAndRulesets(ProcessedArgs args, string configDir, ILo
             string hostUrl = args.GetSetting(SonarProperties.HostUrl);
             serverSettings = null;
 
+            ISonarQubeServer server = this.serverFactory.Create(args);
```

Comment:
The pre-processor no longer needs to know about how to create an IDownloader or an instance of SonarWebService directly. It relies on the factory instead.